### PR TITLE
NAPS-14 - Photos for spots

### PR DESCRIPTION
**Pull request by lukasyelle**
[Ticket](https://ritservices.atlassian.net/browse/NAPS-14?atlOrigin=eyJpIjoiNThlYzUzMjdkYzVhNDczYTk1ZGRmZWRiMTBjNzdjZWYiLCJwIjoiaiJ9)

One of the big reasons that we wanted to rebuild naps was to allow for people to upload images for spots. After a long time, that goal is now realized. 

I tried to keep this as simple as possible, and there really isn't any complicated logic here. Down the road I will be adding the ability to enable / disable photo uploads on a per spot category basis, but that is out of scope currently. 

No tests are added, because I will be restructuring testing for this app soon.

### CODE REVIEW

File: resources/assets/js/components/pages/home/NewSpot.vue [hunk 1]
```diff
@@ -212,6 +235,8 @@
                 }).catch((error) => {
                     console.error(error);
                 });
+
+                this.apiHeaders = window.axios.defaults.headers.common;
```
Comment (lukasyelle) [inline resources/assets/js/components/pages/home/NewSpot.vue]: In order to get the Element file upload [component](https://element.eleme.io/#/en-US/component/upload#user-avatar-upload) working, the proper authentication headers need to be sent. Luckily, they're already created on page load for Axios, so we can just use those for this request.

File: resources/assets/js/pages/home.js [hunk 2]
```diff
@@ -19,6 +19,11 @@ window.coreApiLoaded = (api) => {
  * @return void.
  */
 window.initMap = () => {
+    function verifyPageLoadedProperly(){
```
Comment (lukasyelle) [inline resources/assets/js/pages/home.js]: There was a long standing bug where sometimes the JS would not load properly on the site and that caused it to crash. This is a workaround of that issue

Comment (MontgomeryWatts) [reply resources/assets/js/pages/home.js]: How does this affect the function later in the file (https://github.com/ritstudentgovernment/laravel-naps/blob/a0ee02460089ff7eb6e4016ca3a4937b835e4ac5/resources/assets/js/pages/home.js#L79) that attempts to reload the map if it hangs?

Also the function is set to reload if the map is still loading after 2 seconds. If a user has a slow connection, does this result in the map loading forever?

Comment (lukasyelle) [reply resources/assets/js/pages/home.js]: Im not sure if we need that still, with the solution that I implemented here. Thanks for catching that, ill test it and see if I get any issues.

Comment (lukasyelle) [reply resources/assets/js/pages/home.js]: I have not run into the issue with that secondary check removed. Good catch, ill push up the change

**Pull request merged**